internal/app/auditd/auditd.go

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package auditd registers auditd service and logs audit events.
+package auditd
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/elastic/go-libaudit/v2"
+	"github.com/elastic/go-libaudit/v2/auparse"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime"
+)
+
+// Main is an entrypoint to the auditd service.
+func Main(ctx context.Context, _ runtime.Runtime, logWriter io.Writer) error {
+	return Run(ctx, logWriter)
+}
+
+// Run starts the auditd service.
+//
+// based on https://github.com/elastic/go-libaudit/blob/main/cmd/audit/audit.go
+//
+//nolint:gocyclo
+func Run(ctx context.Context, logWriter io.Writer) error {
+	var writer io.Writer
+
+	client, err := libaudit.NewAuditClient(writer)
+	if err != nil {
+		return fmt.Errorf("[auditd]: failed to create audit client: %w", err)
+	}
+
+	status, err := client.GetStatus()
+	if err != nil {
+		return fmt.Errorf("[auditd]: failed to get audit status: %w", err)
+	}
+
+	if status.Enabled == 0 {
+		if err := client.SetEnabled(true, libaudit.WaitForReply); err != nil {
+			return fmt.Errorf("[auditd]: failed to enable audit: %w", err)
+		}
+	}
+
+	if err = client.SetRateLimit(uint32(4096), libaudit.NoWait); err != nil {
+		return fmt.Errorf("[auditd]: failed to set rate limit: %w", err)
+	}
+
+	if err := client.SetBacklogLimit(8192, libaudit.NoWait); err != nil {
+		return fmt.Errorf("[auditd]: failed to set backlog limit: %w", err)
+	}
+
+	if err := client.SetPID(libaudit.NoWait); err != nil {
+		return fmt.Errorf("[auditd]: failed to set audit PID: %w", err)
+	}
+
+	go func(c *libaudit.AuditClient) {
+		for {
+			rawEvent, err := c.Receive(false)
+			if err != nil {
+				continue
+			}
+
+			// Messages from 1300-2999 are valid audit messages.
+			if rawEvent.Type < auparse.AUDIT_USER_AUTH ||
+				rawEvent.Type > auparse.AUDIT_LAST_USER_MSG2 {
+				continue
+			}
+
+			fmt.Fprintf(logWriter, "type=%s msg=%s\n", rawEvent.Type, rawEvent.Data)
```

Comment:
should we auparse it? (I don't know, just leaving it here, can be a follow-up as well)

Reply:
maybe later if someone asks for it, at least this output is parsable by standard linux tools